portfolio/src/main/webapp/script.js

```diff
@@ -13,11 +13,29 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * fetch and add comments to the page.
  */
-async function getData() {
-  const response = await fetch('/data');
-  const data = await response.text();
-
-  document.getElementById('data-container').innerText = data;
+function getMaxNumberOfComments() {
+  fetch('/data').then(response=>response.json()).then((data)=>{
+      let numComments = document.getElementById('num-comments').value;
+      const comments = document.getElementById('data-container');
+      comments.innerHTML = '';
+      
+      //display certain number of comments by comparing the number of current 
+      //comments with the number of comments requested by the user.
+      for (let i=0; i < numComments && i < data.length; i++){
```

Comment:
Formatting, (i=0, i < numComments, i < data.length) (notice the spaces)

Also ){

Reply:
should I get rid of the spaces or just make it all consistent?

Reply:
Personally I like for (i = 0; i < numComments; i < data.length) {

but consistency is more important.